client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherRequestConvertersTests.java

```diff
@@ -177,4 +188,90 @@ public void testWatcherStatsRequest() {
         }
         assertThat(request.getEntity(), nullValue());
     }
+
+    public void testExecuteWatchByIdRequest() throws IOException {
+
+        boolean ignoreCondition = randomBoolean();
+        boolean recordExecution = randomBoolean();
+        boolean debug = randomBoolean();
+
+        ExecuteWatchRequest request = ExecuteWatchRequest.byId("my_id");
+        request.setIgnoreCondition(ignoreCondition);
+        request.setRecordExecution(recordExecution);
+        request.setDebug(debug);
+
+        request.setActionMode("action1", ExecuteWatchRequest.ActionExecutionMode.SIMULATE);
+
+        String triggerData = "{ \"entry1\" : \"blah\", \"entry2\" : \"blah\" }";
```

Comment:
are these mandatory? if they can be null pls add some `random()` here. If they cant be null then pls add them to the constructor as final's if possible :)

Reply:
++